default/python/AI/AIstate.py

```diff
@@ -399,8 +399,8 @@ def update_system_status(self):
                 # track old/dead enemy fighters for rating assessments in case not enough current info
                 e_f_dict = old_e_fighters if dead_fleet else cur_e_fighters
                 for stats in ship_stats:
-                    attacks = stats[0]
-                    if attacks:
+                    # log only ships that are armed
+                    if stats[0]:
```

Comment:
it's not  a failing of this commit/PR, but I wanted to note something that comes to mind when viewing this line (but not make a commit now that would conflict with this PR)-- it seems we have not updated this line yet for fighters and the possibility of a pure carrier ship with no DF weapons.  I think this line should probably read
```
  if stats[0] or stats[4]:
```
that further, it would probably be best to somewhere have an ATTACKS_INDEX and FIGHTER_CAPACITY_INDEX variable to refer to rather than a hardcoded 0 and 4.